src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx

```diff
@@ -8,108 +8,107 @@ import Operator from '~lib/api/Operator';
-            try {
-                const contract = getContractByName(EContractName.SETTER);
-                signatureHash = await contract.signer.signMessage(rawDataToValidate);
-                setErrorMessage(['']);
-            } catch (e: any){
-                console.log(`Error message: ${e.message}`);
-                setErrorMessage(['You must confirm the signature request through your wallet']);
-                applicationStore.setIsLoading(false);
-                return;
-            }
-            await Operator.getInstance().updateOperatorMetadata(operator.id, signatureHash, payload).then((response) => {
-                const selectedOperator = myAccountStore.ownerAddressOperators.find((op: any) => op.id === operator.id);
-                applicationStore.setIsLoading(false);
-                for (let key in response) {
-                    operator[key] = response[key];
-                    selectedOperator[key] = response[key];
-                }
-                navigate(config.routes.SSV.MY_ACCOUNT.OPERATOR.META_DATA_CONFIRMATION);
-            }).catch((error: any) => {
-                const { message } = error.response.data.error;
-                if (typeof message !== 'string') {
-                    setErrorMessage(Object.values(message));
-                } else {
-                    setErrorMessage([message]);
-                    applicationStore.setIsLoading(false);
-                }
-            });
-            applicationStore.setIsLoading(false);
+  const submitHandler = async () => {
+    const isNotValidity = metadataStore.validateOperatorMetaData();
+    setButtonDisable(isNotValidity);
+    if (!isNotValidity) {
+      let payload = metadataStore.createMetadataPayload();
+      let rawDataToValidate: any = [];
+      fieldsToValidateSignature.forEach(field => {
+        if (payload[field]) {
+          const newItem =
+            field === FIELD_KEYS.OPERATOR_IMAGE ? `logo:sha256:${sha256(payload[field])}` : payload[field];
+          rawDataToValidate.push(newItem);
         }
-    };
+      });
+      rawDataToValidate = rawDataToValidate.join(',');
+      applicationStore.setIsLoading(true);
+      let signatureHash;
+      try {
+        const contract = getContractByName(EContractName.SETTER);
+        signatureHash = await contract.signer.signMessage(rawDataToValidate);
+        setErrorMessage(['']);
+      } catch (e: any) {
+        console.log(`Error message: ${e.message}`);
+        setErrorMessage(['You must confirm the signature request through your wallet']);
+        applicationStore.setIsLoading(false);
+        return;
+      }
+      const updateOperatorResponse = await Operator.getInstance().updateOperatorMetadata(operator.id, signatureHash, payload);
+      if (updateOperatorResponse.result == HttpResult.SUCCESS) {
+        const selectedOperator = myAccountStore.ownerAddressOperators.find((op: any) => op.id === operator.id);
+        applicationStore.setIsLoading(false);
```

Comment:
here and line 87 - remove it cause you have same function call in line 89

Reply:
V